pkg/api/meta/interfaces.go

```diff
@@ -94,6 +94,28 @@ type MetadataAccessor interface {
 	runtime.ResourceVersioner
 }
 
+type RESTScopeName string
+
+const (
+	RESTScopeNameNamespace RESTScopeName = "namespace"
+	RESTScopeNameRoot      RESTScopeName = "root"
+)
+
+// RESTScope contains the information needed to deal with REST resources that are in a resource hierarchy
+type RESTScope interface {
+	// Name of the scope
+	Name() RESTScopeName
+	// ParamName is the optional name of the parameter that should be inserted in the resource url
+	// If empty, no param will be inserted
```

Comment:
Can you put a TODO here that once v1beta1 and v1beta2 are removed, we should consider removing this flexibility everywhere?